src/main/java/homer/common/DeviceTypesImpl.java

```diff
@@ -0,0 +1,40 @@
+package homer.common;
+
+import java.util.LinkedHashSet;
+import java.util.List;
+import java.util.Set;
+
+
+/**
+    Pre-populated valid device types.
+ */
+public final class DeviceTypesImpl implements DeviceTypes {
+
+    private final Set<String> validTypes;
+    private final Set<String> startingTypes = new LinkedHashSet<>(
+        List.of("LIGHT", "OUTLET", "COUTLET", "LOUTLET", "A/C", "HEATER")
+    );
+
+    /**
+     * Maybe we could pass startingTypes as a parameter here.
+     */
+    public DeviceTypesImpl() {
```

Comment:
```suggestion
    public DeviceTypesImpl(final Set<> startingTypes) {
```

Or whatever startingTypes should be.